File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/argcargvtest_runme.m
argcargvtest

largs={'hi','hola','hello'};
if (mainc(largs) != 3)
  error("bad main typemap");
endif

targs={'hi','hola'};
if (mainv(targs,1) != 'hola')
  error("bad main typemap");
endif

targs={'hi', 'hola'};
if (mainv(targs,1) != 'hola')
  error("bad main typemap");
endif

try
  error_flag = 0;
  mainv('hello',1);
  error_flag = 1;
catch
end_try_catch
if (error_flag)
  error("bad main typemap")
endif


initializeApp(largs);

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/argcargvtest_runme.py
from argcargvtest import *

largs = ['hi', 'hola', 'hello']
if mainc(largs) != 3:
    raise RuntimeError("bad main typemap")

targs = ('hi', 'hola')
if mainv(targs, 1) != 'hola':
    print(mainv(targs, 1))
    raise RuntimeError("bad main typemap")

targs = ('hi', 'hola')
if mainv(targs, 1) != 'hola':
    raise RuntimeError("bad main typemap")

try:
    error = 0
    mainv('hello', 1)
    error = 1
except TypeError:
    pass
if error:
    raise RuntimeError("bad main typemap")


initializeApp(largs)
